code/cfile/cfile.cpp

```diff
@@ -187,56 +184,55 @@ int cfile_init(const char *exe_dir, const char *cdrom_dir)
 {
 	int i;
 
-	// initialize encryption
-	encrypt_init();	
+	encrypt_init();	  /* initialize encryption */
 
-	if ( !cfile_inited ) {
-		char buf[CFILE_ROOT_DIRECTORY_LEN];
+	if (cfile_inited) {
+		return 0;
+	}
 
-		cfile_inited = 1;
+	char buf[CFILE_ROOT_DIRECTORY_LEN];
 
-		memset(buf, 0, CFILE_ROOT_DIRECTORY_LEN);
-		strncpy(buf, exe_dir, CFILE_ROOT_DIRECTORY_LEN - 1);
-		i = strlen(buf);
+	cfile_inited = 1;
```

Comment:
I know that the original code also sets this variable here but could you move this to the end of the function where the cfile system is actually initialized? Otherwise calling this function again after it returned an error code will "succeed" without actually making sure that the cfile system is initialized.